services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/S3NativeClientConfiguration.java

```diff
@@ -68,8 +86,33 @@ public S3NativeClientConfiguration(Builder builder) {
         this.checksumValidationEnabled = builder.checksumValidationEnabled == null || builder.checksumValidationEnabled;
         this.readBufferSizeInBytes = builder.readBufferSizeInBytes == null ?
                                      partSizeInBytes * 10 : builder.readBufferSizeInBytes;
+
+        if (builder.httpConfiguration != null) {
+            this.proxyOptions = resolveProxy(builder.httpConfiguration.proxyConfiguration(), tlsContext).orElse(null);
+            this.connectionTimeout = builder.httpConfiguration.connectionTimeout() == null ? null :
```

Comment:
isn't that just the same as `this.connectionTimeout  = builder.httpConfiguration.connectionTimeout()`

Reply:
🤦🏼‍♀️ yes